### PR TITLE
[JENKINS-68994] Removal of SWF

### PR DESCRIPTION
**Pull request by Wadeck**
First attempt, just running the test suites to see if it breaks anything.
Once it's passing, we can remove the connection.swf.

See [JENKINS-68994](https://issues.jenkins.io/browse/JENKINS-68994).

### Proposed changelog entries

* Remove support for Flash-based cross-domain (XDR) transaction support in YUI 2: Connection Manager.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [n/a] Appropriate autotests or explanation to why this change has no tests
- [n/a] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [n/a] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")` if applicable.
- [n/a] For dependency updates: links to external changelogs and, if possible, full diffs

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/6840"></a>

### CODE REVIEW

**Review by basil — COMMENTED** [review]
A promising start! I think removing [YUI 2: Connection Manager](https://yui.github.io/yui2/docs/yui_2.9.0_full/connection/index.html) is an excellent idea, not only because its support for cross-domain (XDR) transactions uses Flash, but also because any usage of YUI 2 at all is technical debt. Happy to help shepherd this across the finish line. :thumbsup:

As you are surely aware, we will need to make sure that this change does not impact any plugins and either prepare the plugins in advance or make an explicit decision to formally deprecate them. A quick search (not exhaustive) shows usages of YUI 2: Connection Manager in at least these two places:

- https://github.com/jenkinsci/gerrit-trigger-plugin/blob/2fb80fd20ba10b60de78d54167a9037b58314b4f/src/main/webapp/js/server-table.js#L170-L178=
- https://github.com/jenkinsci/elasticbox-plugin/blob/master/src/main/webapp/js/variables.js#L591-L605=

The latter has very low adoption numbers, but the former has about 8,000 installations. Note that I am not sure whether these usages depend on the Flash-based cross-domain (XDR) transactions from `connection-{debug,min}.js` or just the basic HTTP transaction support from `connection_core-{debug,min}.js`, but as of commit 76208b9c75f7e607a960498ac4672da35aeab45f the entire YUI 2: Connection Manager is being removed, so all usages must be considered. If all that is desired is to remove the Flash-based cross-domain (XDR) transaction support, a less invasive change would be to replace `connection-{debug,min}.js` with `connection_core-{debug,min}.js` rather than removing it entirely. This would reduce the risk of regression from "any usage of YUI 2: Connection Manager" to "any usage of the Flash-based cross-domain (XDR) transaction support from YUI 2: Connection Manager".

**Comment by Wadeck** [comment]
Thanks for your interest on this effort 🎉
I do not plan to remove completely the Connection manager as it's used even by other YUI component, like tabview or datasource. My objective is to get rid of the SWF file, which has no interest as it's no longer supported by browsers. And thus will "help" scanner to be smarter.

---

- I checked gerrit and elasticbox, none are using the XDR mode.
- Also, there was htmlunit but in the test package and also, without a callback with xdr.
- Then, from the connection code, there are other direct callers, like storage and connection_core, which could also interact with the XDR mode. I checked them:
  - `container-{debug,min}.js` is providing a callback without xdr
  - `tabview-{debug,min}.js` is providing a callback without xdr
  - `connection_core-{debug,min}.js` is potentially using the xdr mode, depending on `getConnectionObject`, which is used by:
    - `utilities.js` from clif-performance-testing-plugin, but it's only choosing between xhr or file mode (not xdr)
- Finally, `storage` is using directly the `_swf` method, to build `StorageEngineSWF`. The `YAHOO.util.Storage` is used by:
  - `sortable.js`. It explicitely requests to get the HTML5 version, not the SWF one.
  - and `catSortable.js` from categorized-view-plugin. It also explicitely requests to get the HTML5 version, not the SWF one.

IOW, no trace in the ecosystem of any usage of the SWF.

---

Removing the SWF without changing the rest of the code has the following advantages:
- No modification of the YUI library code, meaning that if someone is updating the lib, there is no hidden code to adjust
- If a private plugin is requested to get the SWF, it will lead to a 404 error, which is more easily detectable (like network monitor, server log, etc.) than a JS error that is hidden until someone is opening the DevTool.

_(side note, analysing un-typed JS is so harder compared to Java, I could have made some mistakes)_

**Review by timja — APPROVED** [review]

**Review by NotMyFault — APPROVED** [review]

**Review by basil — APPROVED** [review]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Pull request merged**
